### PR TITLE
Updated to be written in TypeScript and compiled to JavaScript.

### PR DESCRIPTION
**Pull request by bmass02**
I updated this package to be written in TypeScript for the sole purpose of being used with TypeScript. I kept the code as close to the original as possible. This change will allow easy to read comments/docs for each supported status code.

Let me know if you have any questions or if you plan on not merging this PR.

Thanks for the well done package.

### CODE REVIEW

**Comment by prettymuchbryce** [comment]
I appreciate your effort here especially with including all of the comments as well. Unfortunately I don't have any plans to merge this. This is a relatively simple project and I don't see the need to complicate it with transpilation or additional dependencies.

However, if you wanted to create a simple TS definition file I would be open to including that.

**Comment by bmass02** [comment]
That's absolutely understandable. I'll make another commit leaving the bare minimum to have the comments and definition file.

**Comment by bmass02** [comment]
@prettymuchbryce I've updated to only include the definition file and updated the README to include the info about TypeScript.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+/**
+ * @deprecated
+ * Official Documentation @ https://tools.ietf.org/html/rfc7231#section-6.4.6
+ *
+ * Was defined in a previous version of the HTTP specification to indicate that a requested response must be accessed by a proxy. It has been deprecated due to security concerns regarding in-band configuration of a proxy.
+ */
+export declare const USE_PROXY = 305;
+/**
+ * Convert the numeric status code to its appropriate title.
+ * @param statusCode One of the available status codes in this package
```
Comment (prettymuchbryce) [inline index.d.ts]: Do we need to document the return value here (`@returns`)?

**Comment by bmass02** [comment]
@prettymuchbryce Just added the ```@returns```. Let me know if that is what you were looking for.

**Comment by prettymuchbryce** [comment]
Thanks again for your work here.

**Pull request merged**